src/Dfc.CourseDirectory.Core/Validation/PostcodeHelper.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Text.RegularExpressions;
+
+namespace Dfc.CourseDirectory.Core.Validation
+{
+    public static class PostcodeHelper
+    {
+        public static readonly Regex UkPostcodePattern = new Regex(
+            @"^((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))\s?[0-9][A-Za-z]{2})$",
+            RegexOptions.Compiled);
+
+        public static string NormalizePostcode(string postcode)
```

Comment:
This feels like a smell. I'm no fan of "Helper" classes as you know and this Normalize function doesn't just normalize, but validates as well.

I suggest doing something similar to the following (not complete implementation):

```
public class PostCode
{
    private string _postCode;

    public PostCode(string postCode)
    {
        _postCode = postCode;
    }
    
    public static bool TryParse(string input, out PostCode postCode)
    {
        // validation, normalization and implementation
    }
    
    public static bool Validate(string postCode)
    {
    }
}
```

This then at least is a useful, descriptive and reusable entity that's available throughout the system.